packages/styles/src/button.scss

```diff
@@ -5,6 +5,55 @@
 
 $fd-button-icon-font-size: 1rem;
 $fd-button-badge-spacing: 0.25rem;
+$btp-button-block: #{$block-button}--btp;
+
+@mixin base-badge() {
```

Comment:
![image](https://github.com/SAP/fundamental-styles/assets/6586561/62784081-6f1f-41f7-affd-43a53458509b)
I see that there's a lot of collision between `.fd-button .fd-button__badge` and `.fd-button.fd-button--btp .fd-button__badge`. Maybe it's a good idea to introduce css var overrides rather than directly assigning new properties?